metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/indexing_master.py

```diff
@@ -138,6 +138,30 @@ def status(self, env):
     def restart(self, env):
         from params import params
         env.set_params(params)
+        commands = IndexingCommands(params)
+        if params.ra_indexing_writer == 'Solr':
+            # Install Solr schemas
+            try:
+                if not commands.is_solr_schema_installed():
+                    commands.solr_schema_install(env)
+                    commands.set_solr_schema_installed()
+
+            except Exception as e:
+                msg = "WARNING: Solr schemas could not be installed.  " \
+                      "Is Solr running?  Will reattempt install on next start.  error={0}"
+                Logger.warning(msg.format(e))
+        else:
```

Comment:
This assumes we will only ever have Solr and ES as our ra_indexing_writers.  As someone who would love to see that expand, can we make it more explicit?

Reply:
Thanks @JonZeolla , as of now we support only ES and SOLR as our ra_indexing_writers, Shall I go ahead and change the description to make it explicit ?

Reply:
I think he's just saying don't default to Elasticsearch in the else condition.  
```
if solr:
  // do solr
else if elasticsearch:
  // do elasticsearch
else:
  // unexpected condition
```

Reply:
Done. Thanks @nickwallen @JonZeolla.

Reply:
can I get +1 on this if this looks good.

Reply:
Is there ever any danger of overwriting user changes in the schemas with this logic?

Reply:
@ottobackwards we are doing pre check wether the schema/templates installed before installing them through restart.

Reply:
ok